src/plugins/chart_expressions/expression_xy/common/utils/log_datatables.ts

```diff
@@ -45,6 +53,30 @@ export const logDatatables = (
     const logTable = prepareLogTable(layer.table, layerDimensions, true);
     handlers.inspectorAdapters.tables.logDatatable(layer.layerId, logTable);
   });
+  if (annotations) {
+    annotations.layers.forEach((layer) => {
+      const logTable = getLogAnnotationTable(annotations.datatable, layer);
+      handlers.inspectorAdapters.tables.logDatatable(layer.layerId, logTable);
+    });
+  }
+};
+
+const getLogAnnotationTable = (data: Datatable, layer: AnnotationLayerConfigResult) => {
+  const layerDimensions: Dimension[] = [
+    [['label'], strings.getLabelLabel()],
+    [['time'], strings.getTimeLabel()],
+  ];
+  layer.annotations
+    .filter((a): a is QueryPointEventAnnotationOutput => a.type === 'query_point_event_annotation')
```

Comment:
You need to filter the rows by the annotations actually belonging to the current annotation layer. Otherwise they get mixed up:
<img width="355" alt="Screenshot 2022-09-13 at 11 08 15" src="https://user-images.githubusercontent.com/1508364/189861076-1c39d35b-20aa-4220-96f6-6609dbf11bf6.png">
<img width="606" alt="Screenshot 2022-09-13 at 11 08 33" src="https://user-images.githubusercontent.com/1508364/189861084-793bef86-5e7d-4a9c-ae14-ca051f971064.png">
<img width="620" alt="Screenshot 2022-09-13 at 11 08 39" src="https://user-images.githubusercontent.com/1508364/189861086-7ca5d2c0-16d1-47d4-a70c-fba2ba827ad3.png">

Reply:
good catch Joe! Thanks, I'll fix it soon 👌🏼